libsolidity/ast/Types.cpp

```diff
@@ -406,28 +603,52 @@ bool IntegerConstantType::isImplicitlyConvertibleTo(Type const& _convertTo) cons
 			return true;
 		return false;
 	}
+	else if (_convertTo.category() == Category::FixedPoint)
+	{
+		auto targetType = dynamic_cast<FixedPointType const*>(&_convertTo);
+		if (m_value == 0)
+			return true;
+		int forSignBit = (targetType->isSigned() ? 1 : 0);
+		if (m_value > 0)
+		{
+			bool properlyScaledBits = m_scalingFactor <= targetType->fractionalBits() ? 
```

Comment:
It think it should be enough to test whether `m_value * pow(bigint(10), target_type->fractionalBits())` can be converted to the signed/unsigned integer type with `target_type->fractionalBits() + target_type->integerBits()` bits.